src/function/scalar/strftime_format.cpp

```diff
@@ -1360,6 +1360,13 @@ bool StrpTimeFormat::ParseResult::TryToDate(date_t &result) {
 	return Date::TryFromDate(data[0], data[1], data[2], result);
 }
 
+bool StrpTimeFormat::ParseResult::TryToTime(dtime_t &result) {
```

Comment:
Unfortunately we are missing `bool Time::TryFromTime(hh, mm, ss, us, off, result)`, so this will not fail if the time is invalid. I think we need to move this duplicated code into `time.cpp` and add a call to `IsValidTime` at the end. Among other things I'm seeing that the parsing code for timestamps does not accept `24:00:00`, which is valid. This may be more than you bargained for, so unless you want to wade in deeper, this is fine for now.

Reply:
The problem with piggybacking off the timestamp parser is that if there is an offset specified in the timestamp format, then the time can be invalid. That could produce unfortunate and possibly unpleasant results.

Reply:
I'll take a look.

Reply:
Fixed all issues here except `24:00:00`. That will need some more consideration.

Reply:
That turns out to not be a thing. Postgres will cast `24:00:00` from ISO-8601 but not parse it in `to_timestamp` so don't worry about it.